pkg/cloud/vsphere/vsphere_test.go

```diff
@@ -59,3 +70,338 @@ func TestResourcesRenderingSmoke(t *testing.T) {
+					Datastore:      "DS2",
+					ComputeCluster: "C2",
+					Networks:       []string{"N2"},
+					ResourcePool:   "RP2",
+					Folder:         "F2",
+				},
+			},
+			{
+				Name:   "west-1a",
+				Region: "west",
+				Zone:   "west-1a",
+				Server: "test-server",
+				Topology: configv1.VSpherePlatformTopology{
+					Datacenter:     "DC3",
+					Datastore:      "DS3",
+					ComputeCluster: "C3",
+					Networks:       []string{"N3"},
+					ResourcePool:   "RP3",
+					Folder:         "F3",
+				},
+			},
+		}
+		vspherePlatformSpec.FailureDomains = append(vspherePlatformSpec.FailureDomains, failureDomainSpec...)
+		vspherePlatformSpec.VCenters = append(vspherePlatformSpec.VCenters, vcenterSpec)
+	}
+	if !emptyVSpherePlatformSpec {
+		platformSpec.VSphere = vspherePlatformSpec
+	}
+
+	return &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: platform,
+			},
+		},
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: infraCloudConfName,
+				Key:  infraCloudConfKey,
+			},
+			PlatformSpec: platformSpec,
+		},
+	}
+}
+
+func makeNetworkResource(network operatorv1.NetworkType) *configv1.Network {
+	return &configv1.Network{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.NetworkStatus{
+			NetworkType: string(network),
+		},
+		Spec: configv1.NetworkSpec{
+			NetworkType: string(network),
+		},
+	}
+}
+
+func TestCloudConfigTransformer(t *testing.T) {
+	type args struct {
+		source  string
+		infra   *configv1.Infrastructure
+		network *configv1.Network
+	}
+	type testConfig struct {
+		name   string
+		args   args
+		want   string
+		errMsg string
+	}
+
+	var tests []testConfig
+
+	tests = append(tests, testConfig{
+		name: "in-tree-to-external",
+		args: args{
+			source: `
+[Global]
+secret-name = "vsphere-creds"
+secret-namespace = "kube-system"
+insecure-flag = "1"
+
+[Workspace]
+server = "test-server"
+datacenter = "DC1"
+default-datastore = "Datastore"
+folder = "/DC1/vm/folder"
+
+[VirtualCenter "test-server"]
+datacenters = "DC1"`,
+			infra: makeInfrastructureResource(configv1.VSpherePlatformType,
+				false,
+				false,
+				false),
+			network: makeNetworkResource(operatorv1.NetworkTypeOpenShiftSDN),
+		},
+		want: `
+[Global]
+secret-name = "vsphere-creds"
+secret-namespace = "kube-system"
+insecure-flag = "1"
+
+[VirtualCenter "test-server"]
+datacenters = "DC1"`,
+		errMsg: "",
+	})
+
+	tests = append(tests, testConfig{
+		name: "labels-already-exist",
+		args: args{
+			source: `
+[Global]
+secret-name = "vsphere-creds"
+secret-namespace = "kube-system"
+insecure-flag = "1"
+
+[Workspace]
+server = "vcenter.sddc-44-236-21-251.vmwarevmc.com"
+datacenter = "SDDC-Datacenter"
+default-datastore = "WorkloadDatastore"
+folder = "/SDDC-Datacenter/vm/jcallen"
+
+[VirtualCenter "vcenter.sddc-44-236-21-251.vmwarevmc.com"]
```

Comment:
Is this virtual center expected to be removed from resulted config?

Reply:
I changed the test to use more generic names. The `[VirtualCenter]` section should remain. The only section removed will be `[Workspace]`.

Reply:
@lobziik I double checked too, purposefully made the structs unequal to make sure the test worked. 
I probably should add some test that fail.